src/components/events/index.ts

```diff
@@ -0,0 +1,26 @@
+import { RedactorDomChanged, RedactorDomChangedPayload } from './RedactorDomChanged';
+import { BlockChanged, BlockChangedPayload } from './BlockChanged';
+import { BlockHovered, BlockHoveredPayload } from './BlockHovered';
+import { FakeCursorAboutToBeSet, FakeCursorAboutToBeSetPayload } from './FakeCursorAboutToBeSet';
+import { FakeCursorHaveBeenSet, FakeCursorHaveBeenSetPayload } from './FakeCursorHaveBeenSet';
+
+/**
+ * Events fired by Editor Event Dispatcher
+ */
+export {
+  RedactorDomChanged,
+  BlockChanged,
+  FakeCursorAboutToBeSet,
+  FakeCursorHaveBeenSet
+};
+
+/**
+ * Event name -> Event payload
+ */
+export interface EditorEventMap extends Record<string, unknown> {
```

Comment:
Do you really need extends here? It is not [type-safe](https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBASwHY2FAZgQwMbDgUQBMEZp8A3YFAWUzDlFSUIGc4AlYbaQgHhZhRkAcwA0cAK5IA1kggB3JAD44AbwBQcOACEANhGzSAEhEpRghAFxwA5HoPHTaCwAVMAT32ZCNgNyaOCxxSKAARCABbAGEAC0wkYQtrG05CYOhw6LiE1w8vH38te0NY+MSrW2LpUpzCN08Ibz8AgDFMaWAoiSgWaABBACMICRgAFQhtYABlYBhkto6unv6hkfHJmZh6-OatBc7u3qgjTEpJqk359oPl49Pgc6RN7caC9QBfdXVuJAE4TGsRBIZEoNDocAAvGoAkV9IYTGYknAAERVBHOOp5V7I0QwwJpbAhTI1crWZGpdJhSIk3INbw4vFVGkVVFw6rZcovem4rR7a5LI6DYZjCbTWZk-YClbC9ZirZY7l4yWHaAnM7AC7ilHK25qh4ap6zLmEZEffzqTAAbRsvQieCg8UIkTgYCgEDANgAur44AB6X1wXgAWhDcDkDCgbqgQA)

Reply:
removed